### ISSUE TITLE
Some stereos don't ask for "GET_CAPABILITIES 3" so AVRCP does not work.

### ISSUE THREAD

**Issue by tfabris** [tfabris/BlueGigaEmpeg#21]
Some stereos (Jen's Onkyo system is one that I've seen) does not send the "GET_CAPABILITIES 3"
command so I never get the chance to respond with "AVRCP RSP 1 2" response to tell it that I can notify it of track changes. This means the AVRCP stuff never works on the system.

There was a point in time when this once worked on that Onkyo system so there must be a mistake somewhere in my query/reponse systems which would kick this thing into gear so that it does all the AVRCP stuff correctly.  Perhaps it's the Class Of Device or something. Investigate this and see if you can make it work.

**Comment by tfabris** [comment]
Attaching log of conversation with Onkyo system.

[Bug repro - Some stereos do not ask for GET CAPABILITIES 3.txt](https://github.com/tfabris/BlueGigaEmpeg/files/1565174/Bug.repro.-.Some.stereos.do.not.ask.for.GET.CAPABILITIES.3.txt)

**Comment by tfabris** [comment]
Onkyo system the play/pause/FF commands work for brief shining moments when it's the Onkyo stereo doing the connection request as opposed to the BlueGiga doing the connection request.

I had a thing where I'd paired the Onkyo with the BlueGiga a long long time ago and it was still in the list of devices on the Onkyo, But I'd erased pairing on the BlueGiga. No matter, the Onkyo paired right up after I powered it on, since I have this thing in "just works" mode all the time.  (Note: I had just disconnected the bluetooth headset a moment prior, this might have been a factor.)

Immediately after powering on the Onkyo system, it did a RING at my BlueGiga and paired up.

After the Onkyo paired with my BlueGiga, for a short time until I reset the BlueGiga, the Onkyo play/pause features worked on my empeg.  (Track titles still did not work.)

**Comment by tfabris** [comment]
Tried all these different CoDs and they did not fix the problem:
  //  0x200420 = 
  //  - Major service class: Audio
  //  - Major device class:  AudioVideo
  //  - Minor device class:  CarAudio
  // 0x280428 =
  //  - Major service class: Audio and Capturing
  //  - Major device class:  AudioVideo
  //  - Minor device class:  HiFi Audio
  // 0x200428 =
  //  - Major service class: Audio
  //  - Major device class:  AudioVideo
  //  - Minor device class:  HiFi Audio and Car Audio

**Comment by tfabris** [comment]
Tried AVRCP PDU 31 1 and AVRCP PDU 31 2 and those didn't help. They were the wrong direction.

**Comment by tfabris** [comment]
Tried AVRCP PDU 10 3 and it didn't help.

**Comment by tfabris** [comment]
Tried a few variations on sending AVRCP NFY (either interim or changed) and they didn't help since there was no transaction label pair to work from.

**Comment by tfabris** [comment]
Things to think about:
- The onkyo doesn't seem to be connecting in Stereo, there are only two connection channels, one AVRCP and one A2DP, so maybe it isn't getting the stereo channels.
- The onkyo when first pairing gets a security error on channel 3. Maybe that's part of the problem. Here's a transcript.


0000001 --- INQUIRY 30
0000004 Empeg state..................................PLAYING  >
0000011 Empeg state..................................PAUSED   ||
0000002 Empeg state..................................PAUSED   ||
0021954 INQUIRY_PARTIAL 00:09:b0:9c:87:9b 240400
0000002 ------------------------------------
0000001    Pairing buddy device address:    
0000001 00:09:b0:9c:87:9b
0000000 ------------------------------------
0000004 --- PAIR 00:09:b0:9c:87:9b
0000053 INQUIRY 1
0000017 INQUIRY 00:09:b0:9c:87:9b 240400
0004614 PAIR 00:09:b0:9c:87:9b OK
0000002 --- CALL 00:09:b0:9c:87:9b 19 A2DP
0000044 CALL 0
0000699 CONNECT 0 A2DP 19
0000001 --- CALL 00:09:b0:9c:87:9b 17 AVRCP
0000053 CALL 1
0000151 CONNECT 1 AVRCP 17
0000057 CONNECT 2 A2DP 19
0000001 --- A2DP STREAMING START
0000002 --- CALL 00:09:b0:9c:87:9b 17 AVRCP
0000070 CALL 3
0000012 A2DP STREAMING START 0
0000001 --- Sending to empeg: C
0000001 Empeg state..................................PLAYING  >
0000017 AUDIO ROUTE 0 A2DP LEFT RIGHT
0000001 Detected: Pairing Mode is Complete.
0000001 --- SET CONTROL RECONNECT 4800 0 0 7 0 A2DP A2DP AVRCP
            STORECONFIG
0000255  
0000000 --------------------------------------
0000001         Pairing process ended.        
0000000 --------------------------------------
0000001  
0000136 Empeg state..................................PLAYING  >
0000002 Empeg state..................................PLAYING  >
0000003 Empeg state..................................PAUSED   ||
0000374 A2DP CODEC SBC JOINT_STEREO 44100 BITPOOL 2-40
0000026 NO CARRIER 1 ERROR 0
0000028 NO CARRIER 3 ERROR 9003 L2CAP_CONNECTION_REJ_SECURITY
0000110 Empeg state..................................PLAYING  >

**Comment by tfabris** [comment]
Yes and no. I fixed the bug in pairing where I accidentally sent too many CALL 17 AVRCP commands, causing it to try to set up an AVRCP channel while the host stereo was in the middle of trying to establish its second (stereo) A2DP channel. That is what was causing the messages NO CARRIER 1 ERROR 0 and NO CARRIER 3 ERROR 9003 L2CAP_CONNECTION_REJ_SECURITY.

But now I've fixed those, the Onkyo pairs in Stereo correctly, and there are no error messages.

But still no query for GET CAPABILITIES 3 from the Onkyo. Sigh.
